app/controllers/admin/ambassador_task_assignments_controller.rb

```diff
@@ -0,0 +1,44 @@
+class Admin::AmbassadorTaskAssignmentsController < Admin::BaseController
+  include SortableTable
+  layout "new_admin"
+
+  def index
+    @ambassador_task_assignments =
+      Kaminari
+        .paginate_array(sorted_assignments)
+        .page(params.fetch(:page, 1))
+        .per(params.fetch(:per_page, 25))
+  end
+
+  private
+
+  def sortable_columns
+    %w[completed_at organization_name task_title ambassador_name]
+  end
+
+  def sorted_assignments
+    assignments =
+      AmbassadorTaskAssignment
+        .includes(:ambassador, :ambassador_task)
+        .completed
+
+    sort_criterion = params.fetch(:sort, "").to_sym
+    direction = params.fetch(:direction, "").to_sym
+
+    case sort_criterion
+    when :completed_at
+      assignments.reorder(completed_at: direction)
+    when :organization_name
+      assignments
+        .to_a
+        .sort_by!(&:organization_name)
```

Comment:
Why can’t you use a join here, and keep this an active record collection, rather than an array?

Reply:
It's not the case that we can't. It would be a complex join because of the way the data is associated (through an associated ambassador and ambassador memberships), but it's feasible. I'll update.

Reply:
Honestly, with regard to the join on organization name - I don't think it's crucial that it sorts alphabetically. It could be sorted by organization_id. And frankly, you could probably even skip sorting by organization altogether.

The reason I believe having it always be active record associations would be nice is that it's more performant and it will enable us to add additional scopes on. Which we will need to do. For example: searching just for a specific task, just for one user, just for one organization.

Reply:
> It could be sorted by organization_id.
Not if we want the user to be able to sort by the organization's name, which how we're representing orgs in the UI. I'm not sure if there's a better way to represent organizations in the UI, and I don't think it would be intuitive to sort values in the column in a way that's not transparent to the user, but I'm open to suggestions.

The complexity here is added from the fact that a user can have multiple ambassador group memberships, so the following is insufficient:
```
assignments.reorder("organizations.name #{direction}")
```

Making progress though.

> You could probably even skip sorting by organization altogether.
So do we want to drop this feature altogether? I'm taking my cues from #802

> The reason I believe having it always be active record associations would be nice is that it's more performant and it will enable us to add additional scopes on. 

I agree, modulo time-to-deployment :)

Reply:
Write a comment that it would add extra time and doesn't deliver sufficient value 😉 and then ignore it.